File: NatureandCode_step07_P5/ArrowParticle.js
// Creates centered drawn 
// unit arrow particle



class ArrowParticle {


  ArrowParticle() {
  }

  constructor(P) {
  }

  create() {
    noStroke();
    fill(255, 150);
    beginShape();
    vertex(-.5, -.175); 
    vertex(.2, -.175);
    vertex(.2, -.5);
    vertex(.5, 0);
    vertex(.2, .5);
    vertex(.2, .175);
    vertex(-.5, .175);
    endShape(CLOSE);
  }
}

File: NatureandCode_step06_pure_Java/ArrowParticle.java
// Creates centered drawn 
// unit arrow particle


// Requires import of Processing core classes
// when .java suffix is added
import processing.core.*;

class ArrowParticle {

  PApplet pApp;

  ArrowParticle() {
  }

  ArrowParticle(PApplet pApp) {
    this.pApp = pApp;
  }

  void create() {
    pApp.noStroke();
    pApp.fill(255, 150);
    pApp.beginShape();
    pApp.vertex(-.5f, -.175f); // need to explicitly cast to floats now
    pApp.vertex(.2f, -.175f);
    pApp.vertex(.2f, -.5f);
    pApp.vertex(.5f, 0);
    pApp.vertex(.2f, .5f);
    pApp.vertex(.2f, .175f);
    pApp.vertex(-.5f, .175f);
    pApp.endShape(PConstants.CLOSE);
  }
}
